.github/actions/install_requirements/action.yml

```diff
@@ -0,0 +1,51 @@
+name: "install requirements"
+description: "run pip install with requirements and upload resulting requirements"
+inputs:
+  requirements_file:
+    description: "name of requirements file to use and upload"
+    required: true
+  install_options:
+    description: "parameters to pass to pip install"
+    required: true
+  python_version:
+    description: "python version to install"
+    default: "3.x"
+
+runs:
+  using: "composite"
+
+  steps:
+    - name: Setup python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python_version }}
+
+    - name: Pip Install
+      run: |
+        touch ${{ inputs.requirements_file }}
+        # -c uses requirements.txt as a constraints file - see comment below
+        pip install -c ${{ inputs.requirements_file }} ${{ inputs.install_options }}
+      shell: bash
+
+    # This eliminates the class of problems where the requirements being given no
+    # longer match what the packages themselves dictate. E.g. In the rare instance
+    # where I install some-package which used to depend on vulnerable-dependency
+    # but now uses good-dependency (despite being nominally the same version)
+    # pip will install both if given a requirements file with -r
+    - name: Validate requirements file
+      run: diff ${{ inputs.requirements_file }} <(pip freeze --exclude-editable)
```

Comment:
```suggestion
      run: [ -s ${{ inputs.requirements_file }} ] && diff ${{ inputs.requirements_file }} lockfiles/${{ inputs.requirements_file }}
```

Reply:
that's not the answer! but I did get this working.